### PR TITLE
Code refactor, updated Unit Test with JsonHexTag Serializer

### PR DESCRIPTION
**Pull request by srsathish92**
### Description (*)
This PR helps to refactor the code of Block/Cart/Shipping in Checkout Module, utilised JsonHexTag class for replacing PHP json_encode with JsonHexTag option, Also covered the unit test case with JsonHexTag class & data provider.

Path: `app/code/Magento/Checkout/Block/Cart/Shipping.php`

### Manual testing scenarios (*)
`vendor/bin/phpunit -c dev/tests/unit/phpunit.xml.dist app/code/Magento/Checkout/Test/Unit/Block/Cart/ShippingTest.php`

### Questions or comments
If any please let me know your feedback

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Checkout/Block/Cart/Shipping.php [hunk 1]
```diff
@@ -3,54 +3,73 @@
...
      * @throws \RuntimeException
      */
     public function __construct(
-        \Magento\Framework\View\Element\Template\Context $context,
-        \Magento\Customer\Model\Session $customerSession,
-        \Magento\Checkout\Model\Session $checkoutSession,
-        \Magento\Checkout\Model\CompositeConfigProvider $configProvider,
+        Context $context,
+        customerSession $customerSession,
+        checkoutSession $checkoutSession,
```
Comment (lbajsarowicz) [inline app/code/Magento/Checkout/Block/Cart/Shipping.php]: Class names and aliases should be PascalCase

File: app/code/Magento/Checkout/Test/Unit/Block/Cart/ShippingTest.php [hunk 2]
```diff
@@ -3,122 +3,215 @@
...
+        $this->layoutProcessorMock = $this->createMock(LayoutProcessorInterface::class);
+        $this->serializerMock = $this->createMock(JsonHexTag::class);
+        $this->jsonHexTagSerializerMock = $this->createMock(JsonHexTag::class);
+        $this->storeManagerMock = $this->createMock(StoreManagerInterface::class);
         $this->layout = [
             'components' => [
-                'firstComponent' => ['param' => 'value'],
-                'secondComponent' => ['param' => 'value'],
+                'firstComponent' => ['param' => 'value']
             ]
```
Comment (lbajsarowicz) [inline app/code/Magento/Checkout/Test/Unit/Block/Cart/ShippingTest.php]: Extract that array into `private const STUB_PREINITIALIZED_COMPONENTS`

File: app/code/Magento/Checkout/Test/Unit/Block/Cart/ShippingTest.php [hunk 3]
```diff
@@ -3,122 +3,215 @@
...
+            ->method('getStoreManager')
+            ->willReturn($this->storeManagerMock);
 
-        $this->model = new \Magento\Checkout\Block\Cart\Shipping(
-            $this->context,
-            $this->customerSession,
-            $this->checkoutSession,
-            $this->configProvider,
-            [$this->layoutProcessor],
+        $this->block = new Shipping(
```
Comment (lbajsarowicz) [inline app/code/Magento/Checkout/Test/Unit/Block/Cart/ShippingTest.php]: Why don't you use ObjectManager helper that mocks all the required classes itself?

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]

File: app/code/Magento/Checkout/Block/Cart/Shipping.php (hunk 1 above)
Comment (srsathish92) [reply app/code/Magento/Checkout/Block/Cart/Shipping.php]: sure, I will update

File: app/code/Magento/Checkout/Test/Unit/Block/Cart/ShippingTest.php (hunk 2 above)
Comment (srsathish92) [reply app/code/Magento/Checkout/Test/Unit/Block/Cart/ShippingTest.php]: sure, I will update

File: app/code/Magento/Checkout/Test/Unit/Block/Cart/ShippingTest.php (hunk 3 above)
Comment (srsathish92) [reply app/code/Magento/Checkout/Test/Unit/Block/Cart/ShippingTest.php]: is it necessary to add ObjectManager helper?

Comment (lbajsarowicz) [reply app/code/Magento/Checkout/Test/Unit/Block/Cart/ShippingTest.php]: It's highly recommended to initiate classes with ObjectManager helper.

Comment (srsathish92) [reply app/code/Magento/Checkout/Test/Unit/Block/Cart/ShippingTest.php]: @lbajsarowicz Thanks for the feedback, then surely I will update.

File: app/code/Magento/Checkout/Block/Cart/Shipping.php (hunk 1 above)
Comment (srsathish92) [reply app/code/Magento/Checkout/Block/Cart/Shipping.php]: @lbajsarowicz I have updated the changes. please review now.

File: app/code/Magento/Checkout/Test/Unit/Block/Cart/ShippingTest.php (hunk 2 above)
Comment (srsathish92) [reply app/code/Magento/Checkout/Test/Unit/Block/Cart/ShippingTest.php]: @lbajsarowicz I have updated the changes. please review now.

File: app/code/Magento/Checkout/Test/Unit/Block/Cart/ShippingTest.php (hunk 3 above)
Comment (srsathish92) [reply app/code/Magento/Checkout/Test/Unit/Block/Cart/ShippingTest.php]: @lbajsarowicz I have updated the changes. please review now.

Comment (srsathish92) [reply app/code/Magento/Checkout/Test/Unit/Block/Cart/ShippingTest.php]: @lbajsarowicz  I have updated the changes, please review

**Review by lbajsarowicz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-6514](https://jira.corp.magento.com/browse/ENGCOM-6514) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Comment by srsathish92** [comment]
@lbajsarowicz can you please add the labels.

**Comment by lbajsarowicz** [comment]
Added `Test coverage`

**Pull request merged**